class/class-wp-scss.php

```diff
@@ -54,91 +89,93 @@ public function __construct ($scss_dir, $css_dir, $compile_method, $sourcemaps)
    * @access public
    */
   public function compile() {
-    global $scssc, $cache;
-    $cache = WPSCSS_PLUGIN_DIR . '/cache/';
+   
+    $input_files = array();
+
+    // Loop through directory and get .scss file that do not start with '_'
+    foreach(new DirectoryIterator($this->scss_dir) as $file) {
+      if (substr($file, 0, 1) != "_" && pathinfo($file->getFilename(), PATHINFO_EXTENSION) == 'scss') {
+        array_push($input_files, $file->getFilename());
+      }
+    }
 
-    //Compiler - Takes scss $in and writes compiled css to $out file
-    // catches errors and puts them the object's compiled_errors property
-    if (!function_exists( 'compiler' )) {
-      function compiler($in, $out, $instance) {
-        global $scssc, $cache;
+    // For each input file, find matching css file and compile
+    foreach ($input_files as $scss_file) {
+      $input = $this->scss_dir.$scss_file;
+      $outputName = preg_replace("/\.[^$]*/",".css", $scss_file);
+      $output = $this->css_dir.$outputName;
```

Comment:
Add spaces around the `.` for better readability and to maintain code style.
```suggestion
      $input = $this->scss_dir . $scss_file;
      $outputName = preg_replace("/\.[^$]*/", ".css", $scss_file);
      $output = $this->css_dir . $outputName;
```